diff --git a/cmake/onnxruntime_common.cmake b/cmake/onnxruntime_common.cmake
index 0e02ad9daa..de5f857884 100644
--- a/cmake/onnxruntime_common.cmake
+++ b/cmake/onnxruntime_common.cmake
@@ -211,4 +211,8 @@ if (ARM64 OR ARM OR X86 OR X64 OR X86_64)
 endif()
 
+if (NOT WIN32)
+  target_link_libraries(onnxruntime_common PUBLIC nsync::nsync_cpp cpuinfo::cpuinfo)
+endif()
+
 if (NOT onnxruntime_BUILD_SHARED_LIB)
   install(DIRECTORY ${PROJECT_SOURCE_DIR}/../include/onnxruntime/core/common  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime/core)
