diff --git a/cmake/adjust_global_compile_flags.cmake b/cmake/adjust_global_compile_flags.cmake
index e072363417..226034b944 100644
--- a/cmake/adjust_global_compile_flags.cmake
+++ b/cmake/adjust_global_compile_flags.cmake
@@ -362,5 +362,6 @@ endif()
 if (WIN32)
+    string(APPEND CMAKE_CXX_FLAGS " /wd4100 /wd4244")
     # required to be set explicitly to enable Eigen-Unsupported SpecialFunctions
     string(APPEND CMAKE_CXX_FLAGS " -DEIGEN_HAS_C99_MATH")
 elseif(LINUX)
     add_compile_definitions("_GNU_SOURCE")
 endif()
